src/prefect/runner/storage.py

```diff
@@ -165,19 +172,47 @@ def _repository_url_with_credentials(self) -> str:
             elif isinstance(v, SecretStr):
                 credentials[k] = v.get_secret_value()
 
-        formatted_credentials = _format_token_from_credentials(
-            urlparse(self._url).netloc, credentials
-        )
-        if url_components.scheme == "https" and formatted_credentials is not None:
+        return _format_token_from_credentials(urlparse(self._url).netloc, credentials)
+
+    @property
+    def _repository_url_with_credentials(self) -> str:
+        if not self._credentials:
+            return self._url
+
+        url_components = urlparse(self._url)
+
+        if url_components.scheme == "https" and self._formatted_credentials is not None:
             updated_components = url_components._replace(
-                netloc=f"{formatted_credentials}@{url_components.netloc}"
+                netloc=f"{self._formatted_credentials}@{url_components.netloc}"
             )
             repository_url = urlunparse(updated_components)
         else:
             repository_url = self._url
 
         return repository_url
 
+    def _git_config(self) -> List[str]:
+        """
+        Build a git configuration to use when running git commands.
```

Comment:
I'm not sure what other use cases we may have, but any other arbitrary config options can get added here.